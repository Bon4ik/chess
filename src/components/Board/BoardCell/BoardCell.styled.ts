import styled from "styled-components";

import { CELL_TYPE, COLOR_PALETTE } from "utils/const";

interface IProps {
  type?: string;
  isActive?: boolean;
}

export const BoardCellContainer = styled.div`
  width: 100px;
  height: 100px;
  display: flex;
  cursor: pointer;
  align-items: center;
  justify-content: center;
  opacity: ${({ isActive }) => (isActive ? 0.6 : 1)};
  background: ${({ type }: IProps) =>
    type === CELL_TYPE.DARK
      ? COLOR_PALETTE.BOARD_DARK
      : COLOR_PALETTE.BOARD_LIGHT};
`;

export const ChessPiece = styled.span`
  font-size: 65px;
  line-height: 1;
  text-transform: uppercase;
  text-shadow: 0 2px
    ${({ type }: IProps) =>
      type === CELL_TYPE.DARK ? COLOR_PALETTE.WHITE : COLOR_PALETTE.BLACK};
  color: ${({ type }: IProps) =>
    type === CELL_TYPE.DARK ? COLOR_PALETTE.BLACK : COLOR_PALETTE.WHITE};
`;
