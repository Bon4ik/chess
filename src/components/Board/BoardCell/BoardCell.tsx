import React from "react";

import { BoardCellContainer, ChessPiece } from "./BoardCell.styled";

export interface IBoardCell extends IBoardCellEntity {
  onCellClick: (targetCoordinates: number[], chessPieceType?: string) => void;
  isActive: boolean;
}

const BoardCellComponent: React.FC<IBoardCell> = ({
  cellType,
  value,
  chessPieceType,
  coordinates,
  onCellClick,
  isActive,
}) => {
  const onClick = () => {
    onCellClick(coordinates, chessPieceType);
  };

  return (
    <BoardCellContainer isActive={isActive} type={cellType} onClick={onClick}>
      <ChessPiece type={chessPieceType}>{value}</ChessPiece>
    </BoardCellContainer>
  );
};

export default BoardCellComponent;
