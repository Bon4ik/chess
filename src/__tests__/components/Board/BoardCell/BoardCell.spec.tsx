import React from "react";
import renderer from "react-test-renderer";
import BoardCell from "components/Board/BoardCell";

describe("BottomBar", () => {
  it("should renders correctly", () => {
    const givenProps = {
      id: 1,
      cellType: "DARK",
      value: "p",
      chessPieceType: "DARK",
      coordinates: [0, 0],
      onCellClick: jest.fn(),
      isActive: true,
    };
    const tree = renderer.create(<BoardCell {...givenProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
