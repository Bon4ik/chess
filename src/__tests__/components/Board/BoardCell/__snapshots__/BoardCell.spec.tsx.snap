// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomBar should renders correctly 1`] = `
<div
  className="sc-bdnylx ccjdWV"
  onClick={[Function]}
  type="DARK"
>
  <span
    className="sc-gtssRu gDJAzW"
    type="DARK"
  >
    p
  </span>
</div>
`;
